back/engines/free/user_confirmation/app/models/user_confirmation/extensions/user.rb

```diff
@@ -8,24 +8,59 @@ def self.included(base)
             validates :email_confirmation_retry_count, numericality: { less_than_or_equal_to: ENV.fetch('EMAIL_CONFIRMATION_MAX_RETRIES', 5) }
             validates :email_confirmation_code_reset_count, numericality: { less_than_or_equal_to: ENV.fetch('EMAIL_CONFIRMATION_MAX_RETRIES', 5) }
 
-            before_validation :reset_confirmation_code, unless: :email_confirmation_code, if: :email_changed?
+            with_options if: :email_changed?, on: :create do
```

Comment:
Just to be sure, this doesn't need to happen when the email is updated? I'm not completely following anymore 😅

Reply:
Ah yes sorry, my mistake here. These changings are about setting up for the email change, but not doing it yet. The `on: :create` check is meant to be removed at some point